dome.gd

```diff
@@ -0,0 +1,80 @@
+extends Node2D
+
+signal fully_infested
+
+const DOME_SPRITES_PATH = "res://art/dome_sprites"
+const INFESTATION_COUNTDOWN = 30
+
+var infestation_percentage: float = 0.0
+var infestation_stage: InfestationStage = InfestationStage.UNINFESTED
+var infestation_type: Globals.InfestationType = Globals.InfestationType.NONE
+var infestation_rate: float = Globals.base_infestation_rate
+var resource_type: Globals.ResourceType = Globals.ResourceType.NONE
+var is_hidden: bool = false
+var connections: Dictionary = {}
+var dome_sprites = []
+
+enum InfestationStage {UNINFESTED, MINOR, MODERATE, MAJOR, FULL, LOST}
+
+func _ready():
+	# Load the dome sprites into an array for easy access
+	var dir = DirAccess.open(DOME_SPRITES_PATH)
+	if dir:
+		dir.list_dir_begin()
+		var file_name = dir.get_next()
+		while file_name != "":
+			if not dir.current_is_dir() && not file_name.ends_with(".import"):
+				dome_sprites.append(load(DOME_SPRITES_PATH + "/" + file_name))
+			file_name = dir.get_next()
```

Comment:
not sure what the cause is, but Safe, moderate, and major seem to have their sprites mixed up for me

Reply:
I was wondering if that might happen. I like the idea of using the enum for infestation stage, I just wasn't sure if it would work outside of my environment...

Reply:
I think a sprite sheet could resolve this. Would only have to load one resource as well.

Reply:
Took different approach with https://github.com/Aveli-games/infestation/pull/2/commits/b1dd8a0bd07e6aaf7e047d4b704497151ea49691

Just used AnimatedSprite2D instead and pick the animation frame based on infestation stage enum value